test/common/session-provider.js

```diff
@@ -0,0 +1,19 @@
+let data;
```

Comment:
Can we use [Sinon mocks](http://sinonjs.org/releases/v2.1.0/mocks/) instead of writing our own code?
It's way easier to inspect the calls and we can even be more precise if required.

Reply:
the trick is that it needs to be required. So I cannot just mock. 

so for verifying data I would have to export one of sinon stubs. 
Doesn't really save lines of code. and will increase complexity

`module.exports.getOptions`
will have code like 

stub.getCalls(0).args[0]

do you insist on sinon approach ?

Reply:
Ah I see, you're right. In that case it does not make sense effectively.

We could use `proxyquire` eventually, but it's fine for now.

Reply:
there was `mock-require` lib, you removed it ))